### PR TITLE
fix: Limit L0 Compaction segment size and count

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #30191

### CODE REVIEW

File: internal/datacoord/compaction_l0_view.go [hunk 1]
```diff
@@ -74,34 +75,75 @@ func (v *LevelZeroSegmentsView) Trigger() (CompactionView, string) {
...
+	curDeltaCount := 0
+	idx := 0
+	for _, view := range segments {
+		targetCount := view.DeltalogCount + curDeltaCount
+		if targetCount <= maxCount || idx == 0 {
+			idx += 1
+			curDeltaCount = targetCount
+			continue
+		}
+		break
```
Comment (congqixia) [inline internal/datacoord/compaction_l0_view.go:120]: NIT: put break(abnormal logic) inside if branch

File: internal/datacoord/compaction_l0_view.go [hunk 2]
```diff
@@ -74,34 +75,75 @@ func (v *LevelZeroSegmentsView) Trigger() (CompactionView, string) {
...
+	var curDeltaSize float64
+	idx := 0
+	for _, view := range segments {
+		targetSize := view.DeltaSize + curDeltaSize
+		if targetSize <= maxSize || idx == 0 {
+			idx += 1
+			curDeltaSize = targetSize
+			continue
+		}
+		break
```
Comment (congqixia) [inline internal/datacoord/compaction_l0_view.go:141]: Ditto

**Comment by XuanYang-cn** [comment]
Ut failed reason: #30441

**Comment by XuanYang-cn** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**